### PR TITLE
Add router to app and refactor DetailCourseView

### PR DESCRIPTION
**Pull request by haiyingweng**
## Overview

Add router to app and refactor DetailCourseView

## Changes Made

- Added a router so we can switch to different components depending on the URL path
- Link DetailCourseView from search result
- Refactored DetailCourseView a little 

## Test Coverage

manual

## Next Steps (delete if not applicable)

- right now if click on a search result, search dropdown doesn't collapse
- backend working on an endpoint to get course by id, will use that once it's implemented

## Screenshots (delete if not applicable)

<details>

  <summary>Screenshot</summary>

https://user-images.githubusercontent.com/48968041/135178657-67802f30-9c37-4f2d-a380-1eb2a282540f.mov

  

</details>

### CODE REVIEW

**Review by Npikielny — APPROVED** [review]
Looks good to me!

File: src/components/DetailCourse/DetailCourseView.tsx [hunk 1]
```diff
@@ -6,43 +6,70 @@ import { Course, Section } from '../../types'
...
-    }
-    [pass, fail] = partition(fail, (s: Section) => s.section.substring(0, 3) == "DIS");
-    if (pass.length > 0) {
-        cards.push(<DetailSectionCard sectionType="Section" sections={pass}/>)
-    }
-    [pass, fail] = partition(fail, (s: Section) => s.section.substring(0, 3) == "LAB");
-    if (pass.length > 0) {
-        cards.push(<DetailSectionCard sectionType="Lab" sections={pass}/>)
+class DetailCourseView extends React.Component<DetailCourseViewProps> {
+
```
Comment (danielVebman) [inline src/components/DetailCourse/DetailCourseView.tsx:25]: Delete newline?

File: src/components/DetailCourse/DetailCourseView.tsx [hunk 2]
```diff
@@ -6,43 +6,70 @@ import { Course, Section } from '../../types'
...
+        return cards
     }
-    if (fail.length > 0) {
-        cards.push(<DetailSectionCard sectionType="Other" sections={fail}/>)
+
+    render() {
+        return (
+            <div className="detail-course-view">
+                <p className="detail-course-view-title-label">{this.props.location.state.title}</p>
+                <div className="section-cards" >
```
Comment (danielVebman) [inline src/components/DetailCourse/DetailCourseView.tsx:67]:
```suggestion
                <div className="section-cards">
```

File: src/components/Search/SearchView.tsx [hunk 3]
```diff
@@ -39,17 +39,23 @@ class SearchView extends React.Component {
...
-        </div>
+          < Link to={{ pathname: `/courses/${course.subjectCode + course.courseNum}`, state: course }} onClick={this.collapseSearch}>
+            <p className="result-text">{`${course.subjectCode} ${course.courseNum}: ${course.title}`}</p>
+          </Link >
+        </div >
       )
     })
     return (
-      <div className="search-container">
+      <div className="search-container" >
```
Comment (danielVebman) [inline src/components/Search/SearchView.tsx:58]:
```suggestion
      <div className="search-container">
```

**Review by danielVebman — APPROVED** [review]
Looks great! 🤩🚢🚀

**Pull request merged**